packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span, Trace } from '../../../types/trace';
+
+// It is a section of span that lies on critical path
+type criticalPathSection = {
+  spanId: string;
+  section_start: number;
+  section_end: number;
+};
+
+export const findRootSpanId = (spans: Span[]): string | undefined => {
+  const rootSpan = spans.find(span => span.references.length === 0 && span.depth === 0);
+  return rootSpan?.spanID;
+};
+
+// This function finds child spans for each span and also sorts childSpanIds based on endTime
+export const findChildSpanIds = (spans: Span[]): Span[] => {
+  const refinedSpanData: Span[] = [];
+  spans.forEach(span => {
+    if (span.hasChildren) {
+      const Children = spans
+        .filter(span2 =>
+          span2.references.some(
+            reference => reference.refType === 'CHILD_OF' && reference.spanID === span.spanID
+          )
+        )
+        .sort((a, b) => b.startTime + b.duration - (a.startTime + a.duration))
+        .map(span2 => span2.spanID);
+      refinedSpanData.push({ ...span, childSpanIds: Children });
+    } else {
+      refinedSpanData.push({ ...span, childSpanIds: [] });
+    }
+  });
+  return refinedSpanData;
+};
+
+// This function finds LFC of a current span which is less than the spawn event
+export const findLastFinishingChildSpanId = (
+  traceData: Trace,
+  currentSpan: Span,
+  spawnTime?: number
+): string | undefined => {
+  if (spawnTime) {
+    return currentSpan.childSpanIds.find(each =>
+      traceData.spans.some(span => span.spanID === each && span.startTime + span.duration < spawnTime)
+    );
+  }
+  return currentSpan.childSpanIds[0];
+};
+
+// This function turncates/drops the overflowing child spans
+export const sanitizeOverFlowingChildren = (spans: Span[]): Span[] => {
+  const sanitizedSpanData: Span[] = [];
+
+  spans.forEach(span => {
+    if (span.references.length) {
+      const parentSpan = span.references.filter(ref => ref.refType === 'CHILD_OF')[0].span!;
+      const childEndTime = span.startTime + span.duration;
+      const parentEndTime = parentSpan.startTime + parentSpan.duration;
+      if (span.startTime >= parentSpan.startTime && childEndTime <= parentEndTime) {
```

Comment:
Let's use switch case, if we have multiple cases?

Reply:
As there are three expressions combinations to compare in each case , Should I also use switch ?? or can we go with if-else

Reply:
I mean, if we have more than 2 cases, it's better to use switch-case, helps to reduce cognitive complexity for understanding code.